TranslationManager/Classes/Manager/TranslatableManager.swift

```diff
@@ -414,7 +421,14 @@ public class TranslatableManager<L: LanguageModel, C: LocalizationModel> {
     /// - Parameter localization: The locale required to request translations for
     /// - Parameter completion: Called when translation fetching has finished. Check if the error
     ///                         object is nil to determine whether the operation was a succes.
-    func updateLocaleTranslations(_ localization: LocalizationModel, completion: ((_ error: Error?) -> Void)? = nil) {
+    func updateLocaleTranslations(_ localization: LocalizationModel, in group: DispatchGroup, completion: ((_ error: Error?) -> Void)? = nil) {
+        
+        group.enter()
+        
+        defer {
```

Comment:
This will almost instantly defer as the scope of this function will exit after executing the async get translations call. This `defer` block needs to go into the get translations completion before the switch :)

Reply:
Good thing you're awake 😄 

Moved inside callback now